app/controllers/avo/application_controller.rb

```diff
@@ -176,7 +176,8 @@ def hydrate_resource
     end
 
     def hydrate_related_resource
-      @related_resource.hydrate(view: action_name.to_sym, user: _current_user, model: @model)
+      model = @related_resource.model_class == @model.class ? @model : nil
```

Comment:
I think this check will fail if the referenced association is of the same model like `Comment` -> `has_many` -> `Comments`. The wrong model will be associated.

We should check **why** is a model hydrated there, and which one.